core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1162,9 +1162,14 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     @throws[Exception]
     def handleNewSession() {
       info("ZK expired; shut down all controller components and try to re-elect")
-      onControllerResignation()
-      inLock(controllerContext.controllerLock) {
-        controllerElector.elect
+      if (controllerElector.getControllerID() != config.brokerId) {
+        onControllerResignation()
+        inLock(controllerContext.controllerLock) {
+          controllerElector.elect
+        }
+      } else {
+        //maybe create by current session or the previous zk session's ephemeral node is not deleted
```

Comment:
@ijuma Thanks for the comment, I can make it more clear if needed.  
There are 2 cases will come to here:
case 1: /contrller ephemeral node is created by current session though other zk event
case 2: /controller ephemeral node is created by previous session, but zk server not delete it yet